### PR TITLE
graphql refactor

### PR DESCRIPTION
**Pull request by kumarabd**
**Description**
- New graphql Middleware to handle function signature
- Provider injector added to graphql resolver
- Handler config added to graphql resolver
- Clear stale data for resync

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -229,28 +229,30 @@ func main() {
...
-	r := router.NewRouter(ctx, h, port)
+	g := graphql.New(graphql.Options{
+		Config:          hc,
+		Logger:          log,
+		MeshSyncChannel: meshsyncCh,
+		BrokerConn:      brokerConn,
+	})
 
+	gp := graphql.NewPlayground(graphql.Options{
+		URL: "/api/system/graphql/query",
```
Comment (leecalcote) [inline cmd/main.go:251]: That's a beautifully crafted endpoint.

**Comment by kumarabd** [comment]
Do not merge before this - https://github.com/meshery/meshery/pull/3567

**Comment by leecalcote** [comment]
> Do not merge before this - #3567

Got it. Will you double-check on this issue/pr reference? It seems to be self-referencing.

**Comment by kumarabd** [comment]
> > Do not merge before this - #3567
> 
> Got it. Will you double-check on this issue/pr reference? It seems to be self-referencing.

Oh sorry, its https://github.com/layer5io/meshkit/pull/87

**Pull request merged**